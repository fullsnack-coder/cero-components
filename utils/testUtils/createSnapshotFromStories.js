import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
