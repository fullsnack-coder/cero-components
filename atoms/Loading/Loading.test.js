import * as stories from './Loading.stories'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

describe('[ atoms / Loading ]', () => {
  describe('#Loading', () => createSnapshotFromStories('Loading', stories))
})
