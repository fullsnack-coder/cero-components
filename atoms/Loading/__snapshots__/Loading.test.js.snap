// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ atoms / Loading ] #Loading when 'Loading' Children is mounted should render 1`] = `
<DocumentFragment>
  Adventures washed awfully web poured pirate IÃ¡ve extend shared? Glows visit bitter fire-breather 400 triumphant splash responsible Oakenshield's prong invitations mortal? It must be taken deep into Mordor and cast back into the fiery chasm from whence it came.
</DocumentFragment>
`;

exports[`[ atoms / Loading ] #Loading when 'Loading' Default is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="loading"
    />
  </div>
</DocumentFragment>
`;
